### ISSUE TITLE
Automatic Push and Pulls

### ISSUE THREAD

**Issue by NepCono** [simonhamp/Gitamic#39]
Hi,

We noticed the Gitamic addon for Statamic and we were wondering, if we can set up automatic **push** and **pulls**.

We are working in Kubernetes and we have more nodes of our backend, we have our code and our content (flat) on 2 different repo's.
When our users are working on the content on one node and it changes we need an automatic push. However all our other nodes should do an automatic pull, because the content has been changed. 

Is this possible ? Looking forward to your reply.

Best regards,
Ronald Buster

**Comment by simonhamp** [comment]
Hi @NepCono, I've intentionally stayed away from automatic push & pull for now as Statamic Pro's own git integration supports automatic push, and pull will always require more configuration than Gitamic would be able to do alone.

For automatic pull, you'd likely need some sort of trigger coming from the node that's done the push or a webhook from your source code tool of choice (e.g. GitHub, Bitbucket etc) that can then instruct the other nodes to do a pull, e.g. the webhook pings a node that knows about all nodes and pings an endpoint in each that runs (or schedules) a `git pull`.

Alternatively, you could have a scheduled task that attempts to pull on a regular basis, but that might be quite risky and may end up pulling quite often when there's nothing to pull (i.e. wasting CPU and network cycles).

But it sounds like you're really talking about a content publishing workflow here. Having had a lot of experience with that in a multi-node setup, I'd highly recommend looking at using a database that is shared between all of the nodes for your content.  I know you'll lose the lovely 'content in version control' aspect of Statamic, but trust me, you'll have a much easier ride.

With the [Eloquent Driver](https://github.com/statamic/eloquent-driver) and built-in Revisions, Statamic has great support for using a database, and I can confirm it's a much better solution for this kind of setup.

Basically, the answer here for now is that you won't need Gitamic to achieve either automatic push or pull, you should be able to do it using other means that are already at your disposal.

**Comment by NepCono** [comment]
Hi @simonhamp thank you for your quick and honest reply.
Much apprechiated 🥇
